src/test/java/core/basesyntax/service/TransactionProcessorTest.java

```diff
@@ -0,0 +1,139 @@
+package core.basesyntax.service;
+
+import static org.junit.Assert.assertTrue;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.Operation;
+import core.basesyntax.service.impl.TransactionProcessorImpl;
+import core.basesyntax.strategy.OperationHandler;
+import core.basesyntax.strategy.impl.BalanceTransactionImpl;
+import core.basesyntax.strategy.impl.PurchaseTransactionImpl;
+import core.basesyntax.strategy.impl.ReturnTransactionImpl;
+import core.basesyntax.strategy.impl.SupplyTransactionImpl;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.junit.Before;
+import org.junit.Test;
+
+public class TransactionProcessorTest {
+    private static final String TEST_FILE_HEADER = "fruit,quantity";
+    private final Map<String, OperationHandler> operationHandlerMap = new HashMap<>();
+
+    @Before
+    public void setUp() {
+        operationHandlerMap.put(Operation.BALANCE.getOperation(), new BalanceTransactionImpl());
+        operationHandlerMap.put(Operation.PURCHASE.getOperation(), new PurchaseTransactionImpl());
+        operationHandlerMap.put(Operation.SUPPLY.getOperation(), new SupplyTransactionImpl());
+        operationHandlerMap.put(Operation.RETURN.getOperation(), new ReturnTransactionImpl());
+    }
+
+    @Test
+    public void process_BalanceOperation_OK() {
+        List<String> testListBalance = new ArrayList<>();
+        testListBalance.add(TEST_FILE_HEADER);
+        testListBalance.add("b,banana,60");
+        Storage storage = new Storage();
+        TransactionProcessor transactionProcessor
+                = new TransactionProcessorImpl(operationHandlerMap);
```

Comment:
same, move creation of common instances in @BeforeAll method

Reply:
Done